src/platform/android/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -73,6 +73,32 @@ CHIP_ERROR DeviceInstanceInfoProviderImpl::GetProductLabel(char * buf, size_t bu
     return Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_ProductLabel, buf, bufSize, dateLen);
 }
 
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+    CHIP_ERROR err;
+    size_t deviceNameSize = 0;
+    char androidDeviceName[ConfigurationManager::kMaxDeviceNameLen];
+    MutableCharSpan androidDeviceNameSpan(androidDeviceName);
```

Comment:
So... androidDeviceNameSpan is not actually used in practice.  Why is it needed?  The uses of its data() and size() below can just as easily be `androidDeviceName` and `sizeof(androidDeviceName)` respectively...